features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/voicemessages/timeline/VoiceMessageMediaRepo.kt

```diff
@@ -88,14 +87,15 @@ class DefaultVoiceMessageMediaRepo @AssistedInject constructor(
             source = mediaSource,
             mimeType = mimeType,
             body = body,
+            useCache = false,
         ).mapCatching {
-            val dest = cachedFilePath.apply { parentFile?.mkdirs() }
-            // TODO By not closing the MediaFile we're leaking the rust file handle here.
-            // Not that big of a deal but better to avoid it someday.
-            if (it.toFile().renameTo(dest)) {
-                dest
-            } else {
-                error("Failed to move file to cache.")
+            it.use { mediaFile ->
+                val dest = cachedFilePath.apply { parentFile?.mkdirs() }
+                if (mediaFile.persist(dest.path)) {
```

Comment:
I understand that the File will stay in the cache directory forever (i.e. the app will never delete it, but it can be deleted if the cache dir is emptied), and in the clear format (i.e. not encrypted). This can be seen as a security vulnerability.

The Rust SDK could do the same and decrypt again on demand if necessary. The rational:

> This allows to close the MediaFile handle after use and keeping the file.

is fine, but we need to delete the file at some point.

The SDK was deleting the file if the MediaFile was closed?

Reply:
> The SDK was deleting the file if the MediaFile was closed?

Yes

Reply:
> This can be seen as a security vulnerability.

EA was emptying this dir on every app startup, we could do the same, or rely on the OS to purge the cache dir when it needs disk space.
We forwarded this question to the PMs though as we'd like to understand what are the security requirements around this feature.

Reply:
https://matrix.to/#/!kCCQTCfnABLKGGvQjo:matrix.org/$fwaijR8CYDFv8AM0RxvHrUPpwPCIekxEdoFZrdGVFcw?via=matrix.org&via=element.io&via=flipdot.org